src/robustbase.py

```diff
@@ -40,117 +40,257 @@ def bias_corr(n):
     return corr_factor
 
 def median(x, low=False, high=False):
-    if low==True:
+    """
+    Calculate the median of a given list of values.
+
+    Parameters
+    ----------
+    x : list-like
+        List of numeric values.
+    low : bool, optional
+        If True, calculate the 'lo-median' (for even sample size, take the smaller of the two middle values).
+    high : bool, optional
+        If True, calculate the 'hi-median' (for even sample size, take the larger of the two middle values).
+
+    Returns
+    -------
+    median : float
+        The calculated median based on the provided options.
+
+    Notes
+    -----
+    The median is a measure of central tendency that represents the middle value in a sorted dataset.
+    It is robust to outliers and is often used as an alternative to the mean.
+
+    References
+    ----------
+    [1] https://docs.python.org/3/library/statistics.html#statistics.median
+    """
+
+    # Check if low flag is set
+    if low == True:
```

Comment:
Can you check this with a linting tool? I recommend `ruff` in Visual Studio code and PyCharm comes with linting by default (it will also uncover that `import math` is unused).
`if low` would be the more pythonic way to go.